cou/utils/juju_utils.py

```diff
@@ -474,17 +468,9 @@ async def _run_update_status_hook(self, unit_name: str) -> bool:
 
         :param unit_name: Name of the unit to run update-status hook
         :type unit_name: str
-        :return: True if command succeeded
-        :rtype: bool
         :raises CommandRunFailed: When update-status hook failed
         """
-        try:
-            await self.run_on_unit(unit_name, "hooks/update-status")
-        except CommandRunFailed as e:
-            if "No such file or directory" in str(e):
-                return False
-            raise e
-        return True
+        await self.run_on_unit(unit_name, "hooks/update-status")  # pragma: no cover
```

Comment:
Why is coverage disabled for this line?  We can cover this line using a unit test, even if we mock run_on_unit.

Reply:
I think because we need to mock the thin wrapper to not affect the next try-except. Mocking the `run_on_unit` will affect the next check, and end up with the problem https://github.com/canonical/charmed-openstack-upgrader/pull/475#discussion_r1663646620. But it's really not point of adding a test for this I guess? If the `run_on_unit` is already tested in other place

Reply:
Yeah probably not much point, but we can add a simple test to ensure coverage at least.  Something like:

```
# pseudocode
def test_dispatch_status(model):
  model.run_on_unit = mock
  model._hook_update_status_hook("unit/1")
  model.run_on_unit.assert_single_call("unit/1", "hooks/update-status")
 
```

Reply:
added